src/main/kotlin/apply/application/EvaluationTargetService.kt

```diff
@@ -0,0 +1,137 @@
+            evaluationRepository.findByIdOrNull(evaluationId) ?: throw IllegalArgumentException()
+        val isSavedBefore: Boolean = evaluationTargetRepository.existsByEvaluationId(evaluationId)
+
+        return when {
+            !evaluation.hasBeforeEvaluation() && !isSavedBefore -> saveEvaluationTargetOfFirst(evaluation)
+            evaluation.hasBeforeEvaluation() && !isSavedBefore -> saveEvaluationTarget(evaluation)
+            !evaluation.hasBeforeEvaluation() && isSavedBefore -> updateEvaluationTargetOfFirst(evaluation)
+            evaluation.hasBeforeEvaluation() && isSavedBefore -> updateEvaluationTarget(evaluation)
+            else -> throw IllegalStateException("평가 대상자 리스트를 불러올 수 없습니다.")
+        }
+    }
+
+    private fun saveEvaluationTargetOfFirst(evaluation: Evaluation): List<EvaluationTarget> {
+        val evaluationTargets: List<EvaluationTarget> = createEvaluationTargetsFromFirst(evaluation)
+
+        return evaluationTargetRepository.saveAll(evaluationTargets)
+    }
+
+    private fun createEvaluationTargetsFromFirst(evaluation: Evaluation): List<EvaluationTarget> {
+        val applicantIds: List<Long> =
+            applicationFormRepository.findByRecruitmentId(evaluation.recruitmentId).map { it.applicantId }
+
+        return applicantRepository.findAllById(applicantIds)
+            .filter { isNotCheater(it.id) }
+            .map { EvaluationTarget(evaluationId = evaluation.id, applicantId = it.id) }
+    }
+
+    private fun isNotCheater(applicantId: Long) = !cheaterRepository.existsByApplicantId(applicantId)
+
+    private fun saveEvaluationTarget(evaluation: Evaluation): List<EvaluationTarget> {
+        val evaluationTargets = createEvaluationTargetsFrom(evaluation)
+
+        return evaluationTargetRepository.saveAll(evaluationTargets)
+    }
+
+    private fun createEvaluationTargetsFrom(evaluation: Evaluation): List<EvaluationTarget> {
+        return evaluationTargetRepository.findByEvaluationId(evaluation.beforeEvaluationId)
+            .filter { it.isSameStatusWith(PASS) }
+            .filter { isNotCheater(it.applicantId) }
+            .map { EvaluationTarget(evaluationId = evaluation.id, applicantId = it.applicantId) }
+    }
+
+    private fun updateEvaluationTargetOfFirst(evaluation: Evaluation): List<EvaluationTarget> {
+        val persistApplicantIds: Set<Long> =
+            evaluationTargetRepository.findByEvaluationId(evaluation.id).map { it.applicantId }.toSet()
+
+        val newApplicantIds: Set<Long> = createEvaluationTargetsFromFirst(evaluation).map { it.applicantId }.toSet()
+
+        update(persistApplicantIds, newApplicantIds, evaluation)
+
+        return evaluationTargetRepository.findByEvaluationId(evaluation.id)
+    }
+
+    private fun update(
+        persistApplicantIds: Set<Long>,
+        newApplicantIds: Set<Long>,
+        evaluation: Evaluation
+    ) {
+        val deletingApplicantIds = persistApplicantIds subtract newApplicantIds
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        val deletingApplicantIds = persistApplicantIds - newApplicantIds
```

Reply:
피드백을 반영하여 수정했습니다. 
가독성이 좋아지네요..!

Reply:
연산자 오버로딩 👍